.cirrus/cirrus_sonar_analysis_npm

```diff
@@ -14,34 +13,101 @@ if [ -n "${GITHUB_BASE_BRANCH}" ]; then
-      -Dsonar.analysis.prNumber=$PULL_REQUEST
-else
-  echo "======= Analyze branch ${GITHUB_BRANCH}"
+      -Dsonar.projectKey="$SONAR_PROJECT_KEY" \
+      -Dsonar.organization="$SONAR_ORGANIZATION" \
+      -Dsonar.projectName="$SONAR_PROJECT_NAME" \
+      -Dsonar.projectVersion="$PROJECT_VERSION" \
+      -Dsonar.host.url="$SONARQUBE_NEXT_URL" \
+      -Dsonar.token="$SONARQUBE_NEXT_TOKEN" \
+      -Dsonar.analysis.buildNumber="$BUILD_NUMBER" \
+      -Dsonar.analysis.pipeline="$CIRRUS_BUILD_ID" \
+      -Dsonar.analysis.sha1="$GIT_SHA1"  \
+      -Dsonar.analysis.repository="$GITHUB_REPO"
+
+elif [[ "${GITHUB_BRANCH}" == "branch-"* ]] && [ "$PULL_REQUEST" == "false" ]; then
+  # analyze maintenance branches as long-living branches
+
+  # Fetch all commit history so that SonarQube has exact blame information
+  # for issue auto-assignment
+  # This command can fail with "fatal: --unshallow on a complete repository does not make sense"
+  # if there are not enough commits in the Git repository
+  # For this reason errors are ignored with "|| true"
+  git fetch --unshallow || true
 
   git fetch origin "${GITHUB_BRANCH}"
 
+  # get current version from pom
+  if [[ $PROJECT_VERSION =~ "-SNAPSHOT" ]]; then
+    echo "======= Found SNAPSHOT version ======="
+    PROJECT_VERSION=$(echo "$PROJECT_VERSION" | sed "s|-SNAPSHOT|+$BUILD_NUMBER|g")
+  else
+    echo "======= Found RELEASE version ======="
+  fi
+
+  npx sonar-scanner \
+      -Dsonar.projectKey="$SONAR_PROJECT_KEY" \
+      -Dsonar.organization="$SONAR_ORGANIZATION" \
+      -Dsonar.projectName="$SONAR_PROJECT_NAME" \
+      -Dsonar.projectVersion="$PROJECT_VERSION" \
+      -Dsonar.host.url="$SONARQUBE_NEXT_URL" \
+      -Dsonar.token="$SONARQUBE_NEXT_TOKEN" \
+      -Dsonar.analysis.buildNumber="$BUILD_NUMBER" \
+      -Dsonar.analysis.pipeline="$CIRRUS_BUILD_ID" \
+      -Dsonar.analysis.sha1="$GIT_SHA1"  \
+      -Dsonar.analysis.repository="$GITHUB_REPO" \
+      -Dsonar.branch.name="$GITHUB_BRANCH"
+
+elif [ "$PULL_REQUEST" != "false" ]; then
+  echo '======= Analyze pull request'
+
+  # Do not analyze a SNAPSHOT version but the version related to this build and PR (includes build number)
+  PROJECT_VERSION=$(echo "$PROJECT_VERSION" | sed "s|-SNAPSHOT|+$BUILD_NUMBER|g")
+
   npx sonar-scanner \
-      -Dsonar.projectKey="org.sonarsource.sonarlint.vscode:sonarlint-vscode" \
-      -Dsonar.organization="sonarsource" \
-      -Dsonar.projectName="SonarLint for VSCode" \
-      -Dsonar.host.url=$SONARQUBE_NEXT_URL \
-      -Dsonar.token=$SONARQUBE_NEXT_TOKEN \
-      -Dsonar.analysis.buildNumber=$BUILD_NUMBER \
-      -Dsonar.analysis.pipeline=$CIRRUS_BUILD_ID \
-      -Dsonar.analysis.sha1=$GIT_SHA1  \
-      -Dsonar.analysis.repository=$GITHUB_REPO
+      -Dsonar.projectKey="$SONAR_PROJECT_KEY" \
+      -Dsonar.organization="$SONAR_ORGANIZATION" \
+      -Dsonar.projectName="$SONAR_PROJECT_NAME" \
+      -Dsonar.projectVersion="$PROJECT_VERSION" \
```

Comment:
I think we don't want this (and this is not done by `regular_mvn_build_deploy_analyze`).